music21/expressions.py

```diff
@@ -325,6 +325,119 @@ def nextMark(self):
         return RehearsalMark(self.nextContent(), numbering=self.numbering)
 
 
+# ------------------------------------------------------------------------------
+class PedalType(common.StrEnum):
+    Sustain = 'sustain'
+    Sostenuto = 'sostenuto'
+    Soft = 'soft'
+    Silent = 'silent'
+
+class PedalForm(common.StrEnum):
+    Line = 'line'
+    Symbol = 'symbol'
+    SymbolAlt = 'symbolalt'
+    SymbolLine = 'symbolline'
+
+class PedalMark(spanner.Spanner):
+    '''
+        A pedal mark spanner contains a pedaled set of notes.
+        The spanner starts at the moment of down-pedal, and
+        ends at the moment of up-pedal.  There can be any
+        number of pedal "bounces" in the middle of the spanner.
+
+        The visual form of the pedal mark can vary.  The following
+        examples use a pedal mark with one "bounce" in the middle::
+
+            Pedal marks can be lines:            |_______^________|
+            Pedal marks can be normal symbolic:  Ped.    * Ped.   *
+            Pedal marks can be altered symbolic: Ped.    Ped.     *
+            Pedal marks can be symbol and line:  Ped.____^________|
+
+        Pedal marks, whether lines, symbols, or a combination, can
+        represent the sustain pedal (Ped.), the sostenuto pedal
+        (Sost.), the soft (una corda) pedal, or (more rarely) the
+        silent (muting) pedal.
+
+        Pedal marks that are symbolic can be abbreviated: e.g. P. instead
+        of Ped., S. instead of Sost.
+
+        Pedal marks that are lines can have non-printed portions
+        (gaps) in them; these are usually started with "simile",
+        but not necessarily.
+    '''
+    def __init__(
+        self,
+        *spannedElements,
+        **keywords
+    ) -> None:
+        super().__init__(*spannedElements, **keywords)
+        from music21 import note
+        self.fillElementTypes = [note.GeneralNote]
+
+        self._pedalType: PedalType|None = None
+        self._pedalForm: PedalForm|None = None
+        self.abbreviated: bool = False
+        self.placement: str|None = None
+        self.startOffset: OffsetQL|None = None  # only needed temporarily by xmlToM21.py
```

Comment:
If this is only needed temporarily, there are two better options:

1. Set an entry in PedalMark.editorial['startOffset'] = ...  (easy)
2. (better) In xmlToM21 create a weakref.WeakKeyDictionary()  (like a Javascript WeakMap) like:  `pedalToStartOffset = weakref.WeakKeyDictionary() ;  pm1 = PedalMark();  pedalToStartOffset[pm1] = 3.0`

There are still some places in the code where a converter is leaving a permanent key on an object, but we're trying to wipe them out.

Reply:
Done.